discovery/discovery.go

```diff
@@ -0,0 +1,42 @@
+package discovery
+
+type Peer struct {
+	Address string
+	IsSelf  bool
+}
+
+// T defines the common interface all discovery mechanisms must follow
+type T interface {
+	// TODO: decide what this is
+}
+
+// K8s implements peer discovery via Kubernetes API calls
+type K8s struct {
```

Comment:
This should be in a contrib repo or in a different package. The discovery package should only hold the discovery interface.

Reply:
Agreed